### PR TITLE
Indicate where `std::slice` structs originate from.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libcore/slice.rs [hunk 1]
```diff
@@ -915,6 +917,9 @@ macro_rules! make_mut_slice {
 ///     println!("{}", element);
 /// }
 /// ```
+///
+/// [`iter`]: ../primitive.slice.html#method.iter
```
Comment (ollie27) [inline src/libcore/slice.rs]: These need to be like `../../std/primitive.slice.html#method.iter` so the links work in the `core` docs as well.

Comment (frewsxcv) [inline src/libcore/slice.rs]: You're right. I forgot the type doesn't also appear in the `core` docs.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**